src/Dapr.PluggableComponents/Adaptors/SingletonComponentProvider.cs

```diff
@@ -0,0 +1,22 @@
+using Grpc.Core;
+
+namespace Dapr.PluggableComponents.Adaptors;
+
+public sealed class SingletonComponentProvider<T> : IDaprPluggableComponentProvider<T>
```

Comment:
I wonder we really need this class or we could get by with a lambda.

Reply:
To do so, we'd need to replace the `IDaprPluggableComponentProvider<T>` interface with a delegate, which would then allow us to use factory methods to return lambdas (especially those that need to capture state, like the multiplexed provider).  That said, the interface type is exposed via the gRPC adaptor constructors, likely injected via DI.  DI can sometimes not play well with lambda types (generally built to register interfaces and concrete types); I'd have to look into how ASP.NET works with them.

We could meet in the middle and keep the interface (for DI and simpler end-user purposes) and just have a generic implementation that accepts a lambda that's called by the interface method.

Reply:
Looking again, it looks like `SingletonComponentProvider<T>` is no longer used after a recent change; I'll remove it.